lib/code_corps/github/sync/issue/issue.ex

```diff
@@ -1,45 +1,38 @@
-  - create or update the `CodeCorps.Task` for the `CodeCorps.Project` in the
-    matched `CodeCorps.GithubRepo`
-
-  If the sync succeeds, it will return an `:ok` tuple with the created or
-  updated task.
-
-  If the sync fails, it will return an `:error` tuple, where the second element
-  is the atom indicating a reason.
+  Performs work identical to `sync/2`, with the adition of associating the
+  resulting `CodeCorps.GithubIssue` with the specified `CodeCorps.GithubPullRequest`.
   """
-  @spec sync((map -> Multi.t), map) :: Multi.t
-  def sync(%{fetch_issue: issue} = changes, _payload) do
-    sync_multi(changes, issue)
-  end
-  def sync(changes, payload) do
-    sync_multi(changes, payload)
+  @spec sync(map, GithubRepo.t, GithubPullRequest.t) :: Multi.t
+  def sync(%{} = payload, %GithubRepo{} = github_repo, %GithubPullRequest{} = github_pull_request) do
+    Multi.new
+    |> Multi.run(:github_issue, fn _ -> payload |> Sync.Issue.GithubIssue.create_or_update_issue(github_repo, github_pull_request) end)
+    |> Multi.run(:issue_user, fn %{github_issue: github_issue} -> Sync.User.RecordLinker.link_to(github_issue, payload) end)
+    |> Multi.run(:task, fn %{github_issue: github_issue, issue_user: user} -> github_issue |> Sync.Issue.Task.sync_github_issue(user) end)
   end
 
-  @spec sync_multi(map, map) :: Multi.t
-  defp sync_multi(%{repo: github_repo, github_pull_request: github_pull_request}, payload) do
-    do_sync_multi({github_repo, github_pull_request}, payload)
-  end
-  defp sync_multi(%{repo: github_repo}, payload) do
-    do_sync_multi({github_repo, nil}, payload)
-  end
+  @doc ~S"""
+  Performs sync of a github issue payload related to a repo.
 
-  defp do_sync_multi({github_repo, github_pull_request}, payload) do
+  Creates or updates a `CodeCorps.GithubIssue`
+  - a `CodeCorps.GithubUser` is created or updated as part of the process and associated to `CodeCorps.GithubIssue`
+  - the record is associated to the `CodeCorps.GithubRepo`
+  - an 'unregistered' `CodeCorps.User` is created if no record with matching `:github_id` is found
+  """
+  @spec sync(map, GithubRepo.t) :: Multi.t
+  def sync(%{} = payload, %GithubRepo{} = github_repo) do
     Multi.new
-    |> Multi.run(:github_issue, fn _ -> IssueGithubIssueSyncer.create_or_update_issue({github_repo, github_pull_request}, payload) end)
-    |> Multi.run(:issue_user, fn %{github_issue: github_issue} -> UserRecordLinker.link_to(github_issue, payload) end)
-    |> Multi.run(:task, fn %{github_issue: github_issue, issue_user: user} -> github_issue |> IssueTaskSyncer.sync_github_issue(user) end)
+    |> Multi.run(:github_issue, fn _ -> payload |> Sync.Issue.GithubIssue.create_or_update_issue(github_repo) end)
+    |> Multi.run(:issue_user, fn %{github_issue: github_issue} -> Sync.User.RecordLinker.link_to(github_issue, payload) end)
+    |> Multi.run(:task, fn %{github_issue: github_issue, issue_user: user} -> github_issue |> Sync.Issue.Task.sync_github_issue(user) end)
```

Comment:
Some minor repetition here, but we likely want to drop transactions at some point anyway, so I figured we should be ok with this for now.